Frontend_Web_Development_React_Angular_Vue/Vue/Week 4/List rendering_(v-for).md

```diff
@@ -0,0 +1,263 @@
+    </head>
+    <body>
+        <div id="app"></div>
+
+    </body>
+    </html>
+
+**JS**
+
+    import Vue from "vue";
+
+    import App from "./App";
+
+    new Vue({
+        el: "#app",
+        render: h => h(App)
+    });
+
+**App.vue**
+
+    <template>
+        <div class="container">
+            <h3 class="p-3 text-center">Vue.js - Display a list of items with v-for</h3>
+            <table class="table table-striped table-bordered">
+                <thead>
+                    <tr>
+                        <th>Name</th>
+                        <th>Email</th>
+                        <th>Role</th>
+                    </tr>
+                </thead>
+                <tbody>
+                    <tr v-for="user in users" :key="user.id">
+                        <td>{{user.firstName}} {{user.lastName}}</td>
+                        <td>{{user.email}}</td>
+                        <td>{{user.role}}</td>
+                    </tr>
+                </tbody>
+            </table>
+        </div>    
+    </template>
+
+    <script>
+    export default {
+        data() {
+            return {
+                users: [
+                    { firstName: 'Frank', lastName: 'Murphy', email: 'frank.murphy@test.com', role: 'User' },
+                    { firstName: 'Vic', lastName: 'Reynolds', email: 'vic.reynolds@test.com', role: 'Admin' },
+                    { firstName: 'Gina', lastName: 'Jabowski', email: 'gina.jabowski@test.com', role: 'Admin' },
+                    { firstName: 'Jessi', lastName: 'Glaser', email: 'jessi.glaser@test.com', role: 'User' },
+                    { firstName: 'Jay', lastName: 'Bilzerian', email: 'jay.bilzerian@test.com', role: 'User' }
+                ]
+            };
+        }
+    };
+    </script>
+
+
+![image](https://user-images.githubusercontent.com/57637086/140802360-c6b5ea72-7e19-40b7-b6f4-9e1971d18f4c.png)
+
+## Example Vue component that renders the users array
+
+The app component template contains some standard html for a heading and table, and inside the table the tr tag uses the
+v-for Vue directive to loop over the users array and render a table row for each user that includes the user name, email
+and role.
+
+The app component contains the hardcoded users array that is rendered with the v-for directive in template. The users
+array is declared as a data property of the Vue component so the template can see it.
+
+    <template>
+        <div class="container">
+            <h3 class="p-3 text-center">Vue.js - Display a list of items with v-for</h3>
+            <table class="table table-striped table-bordered">
+                <thead>
+                    <tr>
+                        <th>Name</th>
+                        <th>Email</th>
+                        <th>Role</th>
+                    </tr>
+                </thead>
+                <tbody>
+                    <tr v-for="user in users" :key="user.id">
```

Comment:
Similarly, there is nothing with a key of "id" in your "user" object.
Correct this with a valid key or use an index (as because index also works as a key when we don't have any id's).